Unable to run app

Puede ser varios problemas:

1. Build failure for goal not being specified
2. apache-surefire-tests llama a clase TestPrincipal y falla
3. lang.java.Object no es reconocida en clase
4. Algunas entidades no tienen mapeo correcto con anotaciones

Puede ser problema 4. que hace que el resto falla de momento. A mirar las relaciones y anotaciones correctamente

Siguientes intentos para arreglarlo:

1. intentar dar gol definido en el pom.xml - no funciona
2. intentar anular apache test - no funciona
3. intentar remover libreria java en IDE, reañadirlo y hacer maven clean, build y install - no funciona

Stack trace:

```
[INFO] [1;31mBUILD FAILURE[m
[INFO] [1m------------------------------------------------------------------------[m
[INFO] Total time:  0.233 s
[INFO] Finished at: 2025-07-26T18:43:09+02:00
[INFO] [1m------------------------------------------------------------------------[m
[ERROR] No goals have been specified for this build. You must specify a valid lifecycle phase or a goal in the format %3Cplugin-prefix%3E:%3Cgoal%3E or %3Cplugin-group-id%3E:%3Cplugin-artifact-id%3E[:%3Cplugin-version%3E]:
```

<img width="1634" height="307" alt="Image" src="https://github.com/user-attachments/assets/d2888780-42c3-401d-8a96-96db4e3081d2" />

--- a/Food_Punck/pom.xml
+++ b/Food_Punck/pom.xml
@@ -99,7 +99,6 @@
 						<exclude>
 							<groupId>org.projectlombok</groupId>
 							<artifactId>lombok</artifactId>
-							<version>1.18.38</version>
 						</exclude>
 					</excludes>
 				</configuration>

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/config/SecurityConfig.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/config/SecurityConfig.java
@@ -1,34 +1,34 @@
-package org.ironhack.Food_Punck.config;
-
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.http.HttpMethod;
-import org.springframework.security.config.Customizer;
-import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
-import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
-import org.springframework.security.config.http.SessionCreationPolicy;
-import org.springframework.security.web.SecurityFilterChain;
-
-@Configuration
-@EnableWebSecurity
-public class SecurityConfig {
-
-	//// below methods establishes https connection to all calls. will update later when user and store require authentication
-	///  link: https://www.codejava.net/frameworks/spring-boot/fix-websecurityconfigureradapter-deprecated
-	@Bean
-	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-	    http.csrf(AbstractHttpConfigurer::disable)
-	      .authorizeHttpRequests(authorizationManagerRequestMatcherRegistry ->
-	              authorizationManagerRequestMatcherRegistry.requestMatchers(HttpMethod.DELETE).hasRole("ADMIN")
-	                     // .requestMatchers("/admin/**").hasAnyRole("ADMIN")
-	                    //  .requestMatchers("/user/**").hasAnyRole("USER", "ADMIN")
-	                    //  .requestMatchers("/login/**").permitAll()
-	                      .anyRequest().authenticated())
-	      .httpBasic(Customizer.withDefaults())
-	      .sessionManagement(httpSecuritySessionManagementConfigurer -> httpSecuritySessionManagementConfigurer.sessionCreationPolicy(SessionCreationPolicy.STATELESS));
-
-	    return http.build();
-	}
-	
-}
+//package org.ironhack.Food_Punck.config;
+//
+//import org.springframework.context.annotation.Bean;
+//import org.springframework.context.annotation.Configuration;
+//import org.springframework.http.HttpMethod;
+//import org.springframework.security.config.Customizer;
+//import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+//import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+//import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+//import org.springframework.security.config.http.SessionCreationPolicy;
+//import org.springframework.security.web.SecurityFilterChain;
+//
+////@Configuration
+////@EnableWebSecurity
+//public class SecurityConfig {
+//
+//	//// below methods establishes https connection to all calls. will update later when user and store require authentication
+//	///  link: https://www.codejava.net/frameworks/spring-boot/fix-websecurityconfigureradapter-deprecated
+//	@Bean
+//	public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+//	    http.csrf(AbstractHttpConfigurer::disable)
+//	      .authorizeHttpRequests(authorizationManagerRequestMatcherRegistry ->
+//	              authorizationManagerRequestMatcherRegistry.requestMatchers(HttpMethod.DELETE).hasRole("ADMIN")
+//	                     // .requestMatchers("/admin/**").hasAnyRole("ADMIN")
+//	                    //  .requestMatchers("/user/**").hasAnyRole("USER", "ADMIN")
+//	                    //  .requestMatchers("/login/**").permitAll()
+//	                      .anyRequest().authenticated())
+//	      .httpBasic(Customizer.withDefaults())
+//	      .sessionManagement(httpSecuritySessionManagementConfigurer -> httpSecuritySessionManagementConfigurer.sessionCreationPolicy(SessionCreationPolicy.STATELESS));
+//
+//	    return http.build();
+//	}
+//	
+//}

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/config/WebConfig.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/config/WebConfig.java
@@ -1,25 +1,25 @@
-package org.ironhack.Food_Punck.config;
-
-import org.ironhack.Food_Punck.util.CommonConst;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.web.servlet.config.annotation.CorsRegistry;
-import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
-
-@Configuration
-public class WebConfig implements WebMvcConfigurer{
-	
-    @Override
-    public void addCorsMappings(CorsRegistry registry) {
-        registry.addMapping(CommonConst.ADD_MAPPINGS)	// currently "/**" this should following everything /
-                .allowedOrigins(CommonConst.ALLOWED_ORIGINS)  // currently 8080, alter this if port is occupied
-                .allowedMethods(CommonConst.ALLOWED_METHOD_GET,
-                		CommonConst.ALLOWED_METHOD_POST,
-                		CommonConst.ALLOWED_METHOD_PUT,
-                		CommonConst.ALLOWED_METHOD_DELETE,
-                		CommonConst.ALLOWED_METHOD_OPTIONS)
-                .allowedHeaders(CommonConst.ALLOWED_HEADERS) // currently "*"
-                .allowCredentials(CommonConst.ALLOWED_CREDENTIALS) // currently true
-                .maxAge(CommonConst.MAX_AGE_SITE); // currently 3600
-    }
-	
-}
+//package org.ironhack.Food_Punck.config;
+//
+//import org.ironhack.Food_Punck.util.CommonConst;
+//import org.springframework.context.annotation.Configuration;
+//import org.springframework.web.servlet.config.annotation.CorsRegistry;
+//import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+//
+////@Configuration
+//public class WebConfig implements WebMvcConfigurer{
+//	
+//    @Override
+//    public void addCorsMappings(CorsRegistry registry) {
+//        registry.addMapping(CommonConst.ADD_MAPPINGS)	// currently "/**" this should following everything /
+//                .allowedOrigins(CommonConst.ALLOWED_ORIGINS)  // currently 8080, alter this if port is occupied
+//                .allowedMethods(CommonConst.ALLOWED_METHOD_GET,
+////                		CommonConst.ALLOWED_METHOD_POST,
+//                		CommonConst.ALLOWED_METHOD_PUT,
+//                		CommonConst.ALLOWED_METHOD_DELETE,
+//                		CommonConst.ALLOWED_METHOD_OPTIONS)
+//                .allowedHeaders(CommonConst.ALLOWED_HEADERS) // currently "*"
+//                .allowCredentials(CommonConst.ALLOWED_CREDENTIALS) // currently true
+//                .maxAge(CommonConst.MAX_AGE_SITE); // currently 3600
+//    }
+//	
+//}

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/controllers/StoreController.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/controllers/StoreController.java
@@ -1,6 +1,7 @@
 package org.ironhack.Food_Punck.controllers;
 
 import java.util.List;
+import java.util.Optional;
 
 import org.ironhack.Food_Punck.models.Product;
 import org.ironhack.Food_Punck.models.Promotion;
@@ -84,15 +85,16 @@ public class StoreController {
 	
 	@PutMapping(CommonConst.URL_BY_STORE_ID) // base + "/store" + "/idStore/{id}"
 	public ResponseEntity<Store> updateStore(@PathVariable int id, @RequestBody Store storeDetails) {
-		Store updatedStore = this.storeRepo.findById(id).orElse(null);
-		if (updatedStore != null) {
-			updatedStore.setId(storeDetails.getId());
-			updatedStore.setName(storeDetails.getName());
-			updatedStore.setEmail(storeDetails.getEmail());
-			return ResponseEntity.ok(this.storeRepo.save(updatedStore));
-		} else {
-			return ResponseEntity.notFound().build();
-		}
+		Optional<Store> updatedStore = this.storeRepo.findById(id);
+//		if (updatedStore != null) {
+//			updatedStore.setId(storeDetails.getId());
+//			updatedStore.setName(storeDetails.getName());
+//			updatedStore.setEmail(storeDetails.getEmail());
+//			return ResponseEntity.ok(this.storeRepo.save(updatedStore));
+//		} else {
+//			return ResponseEntity.notFound().build();
+//		}
+		return null;
 	}
 	
 	@PutMapping(CommonConst.URL_BY_PRODUCT_ID) // base + "/store" + "/idProduct/{id}"
@@ -152,8 +154,8 @@ public class StoreController {
 	/// All 3 get1 mappings ///
 	
 	@GetMapping(CommonConst.URL_BY_STORE_ID) // base + "/store" + "/idStore/{id}"
-	public ResponseEntity<List<Store>> getStoreByID(@PathVariable int id) {
-		List<Store> store = this.storeRepo.findByIdStore(id);
+	public ResponseEntity<Optional<Store>> getStoreByID(@PathVariable int id) {
+		Optional<Store> store = this.storeRepo.findById(id);
 		if (store.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {
@@ -162,8 +164,8 @@ public class StoreController {
 	}
 	
 	@GetMapping(CommonConst.URL_BY_PRODUCT_ID) // base + "/store" + "/idProduct/{id}"
-	public ResponseEntity<List<Product>> getProductByID(@PathVariable int id) {
-		List<Product> product = this.productRepo.findByIdProduct(id);
+	public ResponseEntity<Optional<Product>> getProductByID(@PathVariable int id) {
+		Optional<Product> product = this.productRepo.findById(id);
 		if (product.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {
@@ -172,8 +174,8 @@ public class StoreController {
 	}
 	
 	@GetMapping(CommonConst.URL_BY_PROMO_ID) // base + "/store" + "/idPromo/{id}"
-	public ResponseEntity<List<Promotion>> getPromoByID(@PathVariable int id) {
-		List<Promotion> promo = this.promoRepo.findByIdPromo(id);
+	public ResponseEntity<Optional<Promotion>> getPromoByID(@PathVariable int id) {
+		Optional<Promotion> promo = this.promoRepo.findById(id);
 		if (promo.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/controllers/UserController.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/controllers/UserController.java
@@ -1,6 +1,7 @@
 package org.ironhack.Food_Punck.controllers;
 
 import java.util.List;
+import java.util.Optional;
 
 import org.ironhack.Food_Punck.models.Product;
 import org.ironhack.Food_Punck.models.Promotion;
@@ -269,8 +270,8 @@ public class UserController {
 	/// All 6 get1 mappings ///
 	
 	@GetMapping(CommonConst.URL_BY_USER_ID) // base + "/user" + "/idUser/{id}"
-	public ResponseEntity<List<User>> getUserByID(@PathVariable int id) {
-		List<User> user = this.userRepo.findByIdUser(id);
+	public ResponseEntity<Optional<User>> getUserByID(@PathVariable int id) {
+		Optional<User> user = this.userRepo.findById(id);
 		if (user.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {
@@ -279,8 +280,8 @@ public class UserController {
 	}
 	
 	@GetMapping(CommonConst.URL_BY_REVIEW_ID) // base + "/user" + "/idReview/{id}"
-	public ResponseEntity<List<Review>> getReviewByID(@PathVariable int id) {
-		List<Review> review = this.reviewRepo.findByIdReview(id);
+	public ResponseEntity<Optional<Review>> getReviewByID(@PathVariable int id) {
+		Optional<Review> review = this.reviewRepo.findById(id);
 		if (review.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {
@@ -289,8 +290,8 @@ public class UserController {
 	}
 	
 	@GetMapping(CommonConst.URL_BY_RESULT_ID) // base + "/user" + "/idResult/{id}"
-	public ResponseEntity<List<ResultCompare>> getResultByID(@PathVariable int id) {
-		List<ResultCompare> result = this.resultRepo.findByIdResult(id);
+	public ResponseEntity<Optional<ResultCompare>> getResultByID(@PathVariable int id) {
+		Optional<ResultCompare> result = this.resultRepo.findById(id);
 		if (result.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {
@@ -299,8 +300,8 @@ public class UserController {
 	}
 	
 	@GetMapping(CommonConst.URL_BY_STORE_ID) // base + "/user" + "/idStore/{id}"
-	public ResponseEntity<List<Store>> getStoreByID(@PathVariable int id) {
-		List<Store> store = this.storeRepo.findByIdStore(id);
+	public ResponseEntity<Optional<Store>> getStoreByID(@PathVariable int id) {
+		Optional<Store> store = this.storeRepo.findById(id);
 		if (store.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {
@@ -309,8 +310,8 @@ public class UserController {
 	}
 	
 	@GetMapping(CommonConst.URL_BY_PRODUCT_ID) // base + "/user" + "/idProduct/{id}"
-	public ResponseEntity<List<Product>> getProductByID(@PathVariable int id) {
-		List<Product> product = this.productRepo.findByIdProduct(id);
+	public ResponseEntity<Optional<Product>> getProductByID(@PathVariable int id) {
+		Optional<Product> product = this.productRepo.findById(id);
 		if (product.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {
@@ -319,8 +320,8 @@ public class UserController {
 	}
 	
 	@GetMapping(CommonConst.URL_BY_PROMO_ID) // base + "/user" + "/idPromo/{id}"
-	public ResponseEntity<List<Promotion>> getPromoByID(@PathVariable int id) {
-		List<Promotion> promo = this.promoRepo.findByIdPromo(id);
+	public ResponseEntity<Optional<Promotion>> getPromoByID(@PathVariable int id) {
+		Optional<Promotion> promo = this.promoRepo.findById(id);
 		if (promo.isEmpty()) {
 			return ResponseEntity.notFound().build();
 		} else {

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Address.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Address.java
@@ -1,5 +1,6 @@
 package org.ironhack.Food_Punck.models;
 
+import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
@@ -16,7 +17,8 @@ public class Address {
 	
 	@Id
 	@GeneratedValue(strategy = GenerationType.AUTO)
-	private int id;
+	@Column(name = "id_Address")
+	private int idAddress;
 	private String street;
 	private String number;
 	private String postCode;

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Product.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Product.java
@@ -6,6 +6,7 @@ import java.util.List;
 import org.ironhack.Food_Punck.util.CommonConst;
 import org.ironhack.Food_Punck.util.ProductType;
 
+import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
@@ -26,6 +27,7 @@ public class Product {
 
 	@Id
 	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "id_Product")
 	private int id;
 	private String name;
 	private ProductType type;

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Promotion.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Promotion.java
@@ -5,6 +5,7 @@ import java.sql.Timestamp;
 import org.ironhack.Food_Punck.util.CommonConst;
 import org.ironhack.Food_Punck.util.PromotionType;
 
+import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
@@ -23,6 +24,7 @@ public class Promotion {
 	
 	@Id
 	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "id_Promotion")
 	private int id;
 	private String name;
 	private PromotionType type;

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Review.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/Review.java
@@ -2,6 +2,7 @@ package org.ironhack.Food_Punck.models;
 
 import org.ironhack.Food_Punck.util.CommonConst;
 
+import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
@@ -20,17 +21,18 @@ public class Review {
 	
 	@Id
 	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "id_Review")
 	private int id;
 	private String title;
 	private String text;
 	private int valueRanking;
 	
 	@ManyToOne
-	@JoinColumn(name = CommonConst.MYSQL_ID_USER, referencedColumnName = CommonConst.MYSQL_ID_STORE) /// reviews.iduser, user.iduser
+	@JoinColumn(name = CommonConst.MYSQL_ID_USER, referencedColumnName = CommonConst.MYSQL_ID_USER) /// reviews.iduser, user.iduser
 	private User user;
 	
 	@ManyToOne
 	@JoinColumn(name = CommonConst.MYSQL_ID_STORE, referencedColumnName = CommonConst.MYSQL_ID_STORE) /// reviews.idstore, store.idstore
-	private Store Store;
+	private Store store;
 	
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/User.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/User.java
@@ -5,6 +5,7 @@ import java.util.List;
 
 import org.ironhack.Food_Punck.util.CommonConst;
 
+import jakarta.persistence.Column;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.GenerationType;
@@ -26,6 +27,7 @@ public class User {
 
 	@Id
 	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "id_User")
 	private int id;
 	private String userName;
 	private String name;

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/UserProducts.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/models/UserProducts.java
@@ -35,7 +35,7 @@ public class UserProducts {
 	private User user;
 	
 	@ManyToOne
-	@JoinColumn(name = CommonConst.MYSQL_ID_PRODUCT, referencedColumnName = CommonConst.MYSQL_ID_PRODUCT) /// reviews.idstore, store.idstore
+	@JoinColumn(name = CommonConst.MYSQL_ID_PRODUCT, referencedColumnName = CommonConst.MYSQL_ID_PRODUCT) /// product.idprodcut, store.idstore
 	private Product product;
 
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/AddressRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/AddressRepository.java
@@ -16,6 +16,5 @@ public interface AddressRepository extends JpaRepository<Address, Integer> {
 	
 	Optional<Address> findByPostCode(String postCode);
 	Optional<Address> findByStreet(String street);
-	List<Address> findByIdAddress(int idAddress);
 
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ProductPromotionRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ProductPromotionRepository.java
@@ -11,16 +11,20 @@ import org.springframework.data.repository.query.Param;
 
 public interface ProductPromotionRepository extends JpaRepository<ProductPromotion, ProductPromotionId> {
 	
-	Optional<ProductPromotion> findByProductId(@Param("id_Producto") int idProducto);
+//	Optional<ProductPromotion> findByProductId(@Param("id_Producto") int idProducto);
 	
-	Optional<ProductPromotion> findByPromotionId(@Param("id_Promotion") int idPromo);
+//	Optional<ProductPromotion> findByPromotionId(@Param("id_Promotion") int idPromo);
 	
 //    @Query("SELECT SUM(p.cantidad) FROM ProductoPromocionPedido p WHERE p.idPedido = :idPedido")
 //    Integer calcularCantidadPedido(@Param("idPedido") int idPedido);
 
-	@Query("SELECT * FROM Products_has_Promotions php "
-			+ "INNER JOIN Products p ON php.id_Product = p.id_Product WHERE php.id_Product = :id_Product"
-			+ "INNER JOIN Promotions pro ON php.id_Promotion = pro.id_Promotion WHERE php.id_Promotion = :id_Promotion")
-	List<ProductPromotionId> findByProductAndPromo(@Param("id_Product") int idProduct, @Param("id_Promotion") int idPromo);
-	 
+//	@Query("SELECT php.id_Product, p.product_name, pro.promotion_name FROM Products_has_Promotions php "
+//			+ "INNER JOIN Products p ON php.id_Product = p.id_Product"
+//			+ "INNER JOIN Promotions pro ON php.id_Promotion = pro.id_Promotion "
+//			+ "WHERE php.id_Product = :id_Product AND php.id_Promotion = :id_Promotion")
+//	List<ProductPromotionId> findByProductPromotionId(@Param("id_Product") int idProduct, @Param("id_Promotion") int idPromo);
+	
+	@Query(value = "SELECT * FROM product_promotion WHERE product_promotion_id = ?1 AND product_id = ?2", nativeQuery = true)
+	List<ProductPromotion> findByProductPromotionId(int productPromotionId, int productId);   
+	
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ProductRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ProductRepository.java
@@ -17,7 +17,5 @@ public interface ProductRepository extends JpaRepository<Product, Integer> {
 
 	Optional<Product> findByName(String name);
 	Optional<Product> findByType(Enum<ProductType> type);
-	List<Product> findByIdProduct(int idProduct);
-	List<Product> findByIdStore(int idStore);
 
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/PromotionRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/PromotionRepository.java
@@ -17,6 +17,5 @@ public interface PromotionRepository extends JpaRepository<Promotion, Integer> {
 
 	Optional<Promotion> findByName(String name);
 	Optional<Promotion> findByType(Enum<PromotionType> type);
-	List<Promotion> findByIdPromo(int idPromo);
 	
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ResultCompareRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ResultCompareRepository.java
@@ -6,6 +6,7 @@ import java.util.Optional;
 import org.ironhack.Food_Punck.models.ResultCompare;
 import org.ironhack.Food_Punck.util.CommonConst;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import jakarta.persistence.Table;
@@ -15,8 +16,9 @@ import jakarta.persistence.Table;
 public interface ResultCompareRepository extends JpaRepository<ResultCompare, Integer> {
 	
 	Optional<ResultCompare> findByUserId(int idUser);
-	List<ResultCompare> findByIdResult(int idResult);
-	List<ResultCompare> findByIdProduct(int idProduct);
-	List<ResultCompare> findByIdUserAndIdProduct(int idUser, int idProduct);
+	List<ResultCompare> findByListProductsId(int idProduct);
+	
+	@Query(value = "SELECT * FROM Result_Comparative_Log WHERE id_user = ?1 AND product_id = ?2", nativeQuery = true)
+	List<ResultCompare> findByUserProductId(int idUser, int idProduct);
 
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ReviewRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/ReviewRepository.java
@@ -15,7 +15,7 @@ import jakarta.persistence.Table;
 public interface ReviewRepository extends JpaRepository<Review, Integer> {
 	
 	Optional<Review> findByTitle(String title);
-	Optional<Review> findByStore(int idStore);
-	List<Review> findByIdReview(int idReview);
+	Optional<Review> findByStoreId(int idStore);
+//	List<Review> findByIdReview(int idReview);
 
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/StoreRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/StoreRepository.java
@@ -6,6 +6,8 @@ import java.util.Optional;
 import org.ironhack.Food_Punck.models.Store;
 import org.ironhack.Food_Punck.util.CommonConst;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 import jakarta.persistence.Table;
@@ -15,8 +17,8 @@ import jakarta.persistence.Table;
 public interface StoreRepository extends JpaRepository<Store, Integer> {
 
 	Optional<Store> findByEmail(String email);
-	Optional<Store> findByAddress(int idAddress);
+	@Query("SELECT s FROM Store s WHERE s.address.idAddress = :idAddress")
+	Optional<Store> findByAddressId(@Param(value = "idAddress") int idAddress);
 	Optional<Store> findByName(String name);
-	List<Store> findByIdStore(int idStore);
 	
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/UserProductRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/UserProductRepository.java
@@ -1,0 +1,25 @@
+package org.ironhack.Food_Punck.repositories;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.ironhack.Food_Punck.models.ResultCompare;
+import org.ironhack.Food_Punck.models.UserProducts;
+import org.ironhack.Food_Punck.util.CommonConst;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import jakarta.persistence.Table;
+
+@Repository
+@Table(name = CommonConst.MYSQL_TABLE_USERS_PRODUCTS)
+public interface UserProductRepository extends JpaRepository<UserProducts, Integer> {
+	
+	Optional<UserProducts> findByUserId(int idUser);
+	List<UserProducts> findByProductId(int idProduct);
+	
+	@Query(value = "SELECT * FROM Users_has_Products WHERE id_user = ?1 AND product_id = ?2", nativeQuery = true)
+	List<UserProducts> findByUserProductId(int idUser, int idProduct);
+
+}

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/UserRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/UserRepository.java
@@ -15,9 +15,8 @@ import jakarta.persistence.Table;
 public interface UserRepository extends JpaRepository<User,Integer> {
 	
 	Optional<User> findByEmail(String email);
-	Optional<User> findByAddress(int idAddress);
+//	Optional<User> findByIdAddress(int idAddress);
 	Optional<User> findByName(String firstName);
 	Optional<User> findByUserName(String userName);
-	List<User> findByIdUser(int idUser);
 
 }

--- a/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/UserStoreRepository.java
+++ b/Food_Punck/src/main/java/org/ironhack/Food_Punck/repositories/UserStoreRepository.java
@@ -5,18 +5,26 @@ import java.util.Optional;
 
 import org.ironhack.Food_Punck.models.UserStore;
 import org.ironhack.Food_Punck.models.UserStoreId;
+import org.ironhack.Food_Punck.util.CommonConst;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
 
+import jakarta.persistence.Table;
+
+@Repository
+@Table(name= CommonConst.MYSQL_TABLE_USERS_STORES)
 public interface UserStoreRepository extends JpaRepository<UserStore, UserStoreId> {
 	
-	Optional<UserStore> findByUserId(@Param("id_User") int idUser);
-	Optional<UserStore> findByStoreId(@Param("id_Store") int idStore);
+	Optional<UserStore> findByIdUser(@Param("id_User") int idUser);
+	Optional<UserStore> findByIdStore(@Param("id_Store") int idStore);
 	
-	@Query("SELECT * FROM Users_follow_Stores ufs"
-			+ "INNER JOIN User u ON ufs.id_User = u.id_User WHERE ufs.id_User = :id_User"
-			+ "INNER JOIN Stores s ON ufs.id_Store = s.id_Store WHERE ufs.id_Store = :id_Store")
-	List<UserStoreId> findByUserAndStoreId(@Param("id_User") int idUser, @Param("id_Store") int idStore);
+//	@Query("SELECT * FROM Users_follow_Stores ufs"
+//			+ "INNER JOIN User u ON ufs.id_User = u.id_User WHERE ufs.id_User = :id_User"
+//			+ "INNER JOIN Stores s ON ufs.id_Store = s.id_Store WHERE ufs.id_Store = :id_Store")
+	
+	@Query(value = "SELECT * FROM Users_follow_Stores WHERE id_user = ?1 AND id_store = ?2", nativeQuery = true)
+	List<UserStoreId> findByUserStoreId(@Param("id_User") int idUser, @Param("id_Store") int idStore);
 
 }

--- a/Food_Punck/src/main/resources/application.properties
+++ b/Food_Punck/src/main/resources/application.properties
@@ -23,4 +23,7 @@ server.ssl.key-store-type=pkcs12
 server.ssl.key-alias=springboot
 server.ssl.key-password=12345678
 server.ssl.port=8443
-server.ssl.enabled=true
+server.ssl.enabled=false
+# remember to change above to true for ssl certificate
+
+server.port=9090

